### DISCUSSION TITLE
Select all edits in *Refactor View* when `WorkspaceEdit` with `WorkspaceEditEntryMetadata.needsConfirmation` set to `true`

### DISCUSSION THREAD

**Discussion by BoykoAlex** [microsoft/vscode-discussions#972] [Extension Development QnA] [answered]
Here is a command snippet:
```typescript
        let disposable = VSCode.commands.registerCommand('workspaceedit-test.helloWorld', () => {
            const edit: VSCode.WorkspaceEdit = new VSCode.WorkspaceEdit();
    
            const folders = VSCode.workspace.workspaceFolders;
            if (!folders || folders.length === 0) {
                VSCode.window.showInformationMessage('Open a folder first');
                return;
            }
            const newFile = VSCode.Uri.joinPath(folders[0].uri, 'foo.txt');
            edit.createFile(newFile, 
                { overwrite: true }, 
                { label: 'New text', needsConfirmation: true }
            );
    
            const initialContent = new TextEdit(new Range(new Position(0,0), new Position(0, 0)), "Hello World\nOne more line");
            edit.set(newFile, [initialContent]);
    
            VSCode.workspace.applyEdit(edit);
        });

```

The expectation is before the edit is applied to see the **Refactoring View** which does show up. However, none of the changes are selected and I'm forced to check-mark edits manually. How can pre-select/check mark all edits?

**Comment by isidorn** [comment]
Not sure if possible. @jrieken should know.

**Comment by BoykoAlex** [comment]
I guess what I'd like is **preview** the changes rather than **confirm** changes...

**Comment by jrieken** [comment, marked as answer]
> I guess what I'd like is preview the changes rather than confirm changes...

Today, extension cannot make that decision because we believe it's up to the user

**Comment by BoykoAlex** [comment]
@jrieken  Well, extension can ask what user wants but then there is no API available to show "Refactor Preview" with all changes selected. Do you think it'd be possible to add something in the VSCode API to allow such a thing?

**Reply by isidorn** [reply]
No current plans to add that.
